exporter/collector/logs.go

```diff
@@ -43,20 +62,149 @@ func (l *LogsExporter) Shutdown(ctx context.Context) error {
 }
 
 func (l *LogsExporter) PushLogs(ctx context.Context, ld pdata.Logs) error {
+	mapper := &metricMapper{} // Refactor metricMapper to map MRs for logging?
+	logPushErrors := []error{}
 	for i := 0; i < ld.ResourceLogs().Len(); i++ {
 		rl := ld.ResourceLogs().At(i)
-		mapper := &metricMapper{} // Refactor metricMapper to map MRs for logging?
 		mr, _ := mapper.resourceToMonitoredResource(rl.Resource())
 		for j := 0; j < rl.InstrumentationLibraryLogs().Len(); j++ {
 			ill := rl.InstrumentationLibraryLogs().At(j)
 			for k := 0; k < ill.LogRecords().Len(); k++ {
 				log := ill.LogRecords().At(k)
-				l.logger.Log(logging.Entry{
-					Resource: mr,
-					Payload:  log.Body().AsString(),
-				})
+				entry, err := l.logToEntry(log, make(map[string]interface{}), mr)
+				if err != nil {
+					logPushErrors = append(logPushErrors, err)
+				} else {
+					l.logger.Log(entry)
+				}
 			}
 		}
 	}
+	if len(logPushErrors) > 0 {
+		return multierr.Combine(logPushErrors...)
+	}
 	return nil
 }
+
+func (l *LogsExporter) logToEntry(
+	log pdata.LogRecord,
+	attributes map[string]interface{},
+	mr *monitoredres.MonitoredResource,
+) (logging.Entry, error) {
+	entry := logging.Entry{
+		Timestamp: log.Timestamp().AsTime(),
+		Severity:  logging.Severity(log.SeverityNumber()),
+		Trace:     log.TraceID().HexString(),
+		SpanID:    log.SpanID().HexString(),
+		Resource:  mr,
+	}
+
+	logBody := log.Body().BytesVal()
+
+	if l.parseHttpRequest {
+		httpRequest, strippedLogBody, err := parseHttpRequest(logBody)
+		if err != nil {
+			l.obs.log.Debug("error parsing HTTPRequest", zap.Error(err))
+		} else {
+			entry.HTTPRequest = httpRequest
+			logBody = strippedLogBody
+		}
+	}
+
+	if len(logBody) > 0 {
+		unmarshalledLogBody := make(map[string]interface{})
+		err := json.Unmarshal(logBody, &unmarshalledLogBody)
+		if err != nil {
+			entry.Payload = unmarshalledLogBody
+		}
+	}
```

Comment:
Not sure what the issue is, but log bodies are coming through empty when I run this (see below). is there a simpler way we can pass this for now or just continue using text?
<img width="456" alt="Screen Shot 2022-04-05 at 12 35 17 PM" src="https://user-images.githubusercontent.com/1839101/161803007-a0184c40-7200-4b04-a413-4b2b08b915f8.png">

Reply:
Fixed in [72b3690](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/344/commits/72b3690d4a49970b7eaffdd10072dab376d4c72e)